mcs/class/Mono.Profiler.Log/Mono.Profiler.Log/LogEventPool.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Mono.Profiler.Log {
+
+	sealed class LogEventPool {
```

Comment:
ad 1, I think that's fine
ad 2, you do at insert
ad 3, you have start point for each event because you read it but not stored it